admin/controllers/check-form.js

```diff
@@ -0,0 +1,269 @@
+const path = require('path')
+const fs = require('fs-extra')
+const CheckForm = require('../models/check-form')
+const checkFormService = require('../services/check-form.service')
+const checkFormDataService = require('../services/data-access/check-form.data.service')
+const checkWindowService = require('../services/check-window.service')
+
+/**
+ * Display landing page for 'test developer' role.
+ * @param req
+ * @param res
+ * @param next
+ * @returns {Promise.<void>}
+ */
+const getTestDeveloperHome = async (req, res, next) => {
+  res.locals.pageTitle = 'MTC for test development'
+  try {
+    req.breadcrumbs(res.locals.pageTitle)
+    res.render('test-developer/test-developer-home', {
+      breadcrumbs: req.breadcrumbs()
+    })
+  } catch (error) {
+    next(error)
+  }
+}
+
+/**
+ * Display initial 'upload and view' check forms page.
+ * @param req
+ * @param res
+ * @param next
+ * @returns {Promise.<*>}
+ */
+const uploadAndViewForms = async (req, res, next) => {
+  res.locals.pageTitle = 'Upload and view forms'
+  try {
+    const forms = await checkFormDataService.getActiveForms().sort({createdAt: -1}).exec()
+    let formData = forms.map(e => { return e.toJSON() })
+    const checkWindows = await checkWindowService.getCheckWindowsAssignedToForms()
+
+    formData.forEach(f => {
```

Comment:
I see you are doing some data processing here, shouldn't that be included in a service in order to keep the controller more lean?

Reply:
@GeorgeChatzigiannis As explained in private and dev chats, "285 is a high level refactoring of check forms, it doesn’t include all the refactoring but the core modifications" - More refactoring of controllers, services, etc is coming next. This PR was getting too big.